File: task_equivalents/llm_translations/starcoder/base/python/data_manipulation_preprocessing/01_impute_encode_scale.py
import pandas as pd
from sklearn.preprocessing import scale

Auto = pd.read_csv('data/csv/Auto.csv', na_values='?')
Auto.dropna(inplace=True)

Auto['mpg01'] = (Auto['mpg'] > Auto['mpg'].median()).astype(int)
Auto = Auto.drop(['mpg', 'name'], axis=1)

var_no_scale = ['cylinders', 'year', 'origin', 'mpg01']
var_to_scale = ['displacement', 'horsepower', 'weight', 'acceleration']
scaled_var = scale(Auto[var_to_scale])
df = pd.DataFrame(scaled_var, columns=var_to_scale)

df['cylinders'] = Auto['cylinders']
df['year'] = Auto['year']
df['origin'] = Auto['origin']
df['mpg01'] = Auto['mpg01']

x = df.drop('mpg01', axis=1)
y = df['mpg01']

File: task_equivalents/llm_translations/starcoder/base/r/data_manipulation_preprocessing/01_impute_encode_scale.R
library(dplyr)
library(caret)
Auto <- read.csv('data/csv/Auto.csv', na.strings='?') %>%
  filter(!is.na(horsepower)) %>% # remove missing values
  mutate(mpg01 = ifelse(mpg > median(mpg), 1, 0),
         cylinders = as.factor(cylinders),
         origin = as.factor(origin))
var_to_scale <- c('displacement', 'horsepower', 'weight', 'acceleration')
preProcValues <- preProcess(Auto[, var_to_scale], method = c("center", "scale"))
scaled_var  <- predict(preProcValues, Auto[, var_to_scale])
df  <- cbind(scaled_var, Auto[, -which(names(Auto) %in% var_to_scale)])
x  <- df[, -(ncol(df)-1)]
y  <- df[, ncol(df)]
